libs/platform/experience/composition/src/composition.component.ts

```diff
@@ -35,6 +36,26 @@ export class CompositionComponent extends LayoutMixin(
   protected registryService = resolve(ComponentsRegistryService);
   protected layoutBuilder = resolve(LayoutBuilder);
 
+  protected $dynamicVisibilityState = signal(
+    this.contentController.dynamicVisibilityState()
+  );
+
+  @elementEffect()
+  protected toggleVisibilityState = (): void => {
```

Comment:
Does this mean that we only support visibility on compositions?

Reply:
For now - yes. I tried co combine all related logic inside content mixin, but @elementEffect in mixin caused strange glitches on ssr. I asked Stan for some help. I agree, would be nice to have more generic solution, that can be applied for any component in the future, but for now composition only